src/__test__/table-theme-colors.spec.js

```diff
@@ -0,0 +1,481 @@
+    spyQuerySelector.mockReturnValue(undefined);
+    spyGetComputedStyle.mockReturnValue(undefined);
+    spyGetStyle.mockReturnValue(undefined);
+  });
+
+  describe('when there is no background color from theme or css file', () => {
+    it('should return the default value', () => {
+      const result = tableThemeColors(theme);
+      expect(result).toEqual(defaultValue);
+    });
+  });
+
+  describe('when there is only background color on sheet', () => {
+    it('should return the default value ', () => {
+      global.document.querySelector = jest.fn((selector) => {
+        if (selector === '.qv-panel-sheet') return selector;
+        return undefined;
+      });
+      global.window.getComputedStyle = jest.fn((selector) => {
+        if (selector === '.qv-panel-sheet') {
+          return { backgroundColor: '#fff' };
+        }
+        return undefined;
+      });
+      let result = tableThemeColors(theme);
+      expect(result).toEqual(defaultValue);
+
+      global.window.getComputedStyle = jest.fn((selector) => {
+        if (selector === '.qv-panel-sheet') {
+          return { backgroundColor: '#000' };
+        }
+        return undefined;
+      });
+      result = tableThemeColors(theme);
+      expect(result).toEqual(defaultValue);
+    });
+  });
+
+  describe('when there is background color on object', () => {
+    describe('when there is only background color on object', () => {
+      it('should return the default value and backgroundColor when the background color is light', () => {
+        global.document.querySelector = jest.fn((selector) => {
+          if (selector === '.qv-object .qv-inner-object') return selector;
+          return undefined;
+        });
+        global.window.getComputedStyle = jest.fn((selector) => {
+          let style;
+          if (selector === '.qv-object .qv-inner-object') {
+            style = { backgroundColor: '#fff' };
+          }
+          return style;
+        });
+        const result = tableThemeColors(theme);
+        expect(result).toEqual({ ...defaultValue, backgroundColor: '#fff' });
+      });
+
+      it('should return the darkBackgroundColorValue and backgroundColor when the background color is dark', () => {
+        global.document.querySelector = jest.fn((selector) => {
+          if (selector === '.qv-object .qv-inner-object') return selector;
+          return undefined;
+        });
+        global.window.getComputedStyle = jest.fn((selector) => {
+          let style;
+          if (selector === '.qv-object .qv-inner-object') {
+            style = { backgroundColor: '#000' };
+          }
+          return style;
+        });
+        const result = tableThemeColors(theme);
+        expect(result).toEqual({
+          ...darkBackgroundColorValue,
+          backgroundColor: '#000',
+        });
+      });
+
+      it('should return the defaultValue and backgroundColor and isBackgroundTransparentColor to be true when the background color is transparent', () => {
```

Comment:
So for all the transparent tests, there is no need to do more than one type of transparent. You already added tests for the isTransparent function, so you can trust that it will return the correct thing

Reply:
Oki